### PR TITLE
[REVIEW] Add support for `UNION`

### PR DESCRIPTION
**Pull request by galipremsagar**
Fixes: #470

This PR adds `UNION` support in `dask-sql` which uses `datafusion`'s `union` logic.

### CODE REVIEW

**Comment by andygrove** [comment]
Related DataFusion issue: https://github.com/apache/arrow-datafusion/issues/2573

File: dask_planner/src/sql/logical/union.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    union: Union,
+}
+
+#[pymethods]
+impl PyUnion {
+    #[pyo3(name = "all")]
+    pub fn all(&mut self) -> PyResult<bool> {
+        println!("{:?}", self.union.inputs[0]);
+        println!("{:?}", self.union.schema.metadata());
+        println!("{:?}", self.union.alias);
```
Comment (jdye64) [inline dask_planner/src/sql/logical/union.rs:18]: `println!` statements should be removed. If you want you could implement `__str__` impl like 

```
fn __repr__(&self) -> &'static str {
        format!("{:?}\n{:?}\n{:?}", 
                        self.union.inputs[0],
                        self.union.schema.metadata(),
                        self.union.alias)
}
```

File: dask_planner/src/sql/logical/union.rs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    }
+}
+
+impl TryFrom<LogicalPlan> for PyUnion {
+    type Error = PyErr;
+
+    fn try_from(logical_plan: LogicalPlan) -> Result<Self, Self::Error> {
+        match logical_plan {
+            LogicalPlan::Union(union) => Ok(PyUnion { union }),
+            _ => panic!("something went wrong here"),
```
Comment (jdye64) [inline dask_planner/src/sql/logical/union.rs:29]: Can replace this with what the other implementations are using. It isn't that great either really. Need to come up with something more clear error wise later.

```
Err(py_type_err("unexpected plan"))
```

**Review by jdye64 — CHANGES_REQUESTED** [review]
Few small changes. LGTM after that. Thanks Prem!

File: dask_planner/src/sql/logical/union.rs (hunk 2 above)
Comment (galipremsagar) [reply dask_planner/src/sql/logical/union.rs]: Done 👍

File: dask_planner/src/sql/logical/union.rs (hunk 1 above)
Comment (galipremsagar) [reply dask_planner/src/sql/logical/union.rs]: Thanks Jeremy, disabled this method for future use along with a TODO.

**Review by jdye64 — APPROVED** [review]

**Review by ayushdg — COMMENTED** [review]
Changes lgtm! We should update #470 to track that union all is still pending on upstream datafusion changes.

**Pull request merged**